core/src/main/java/com/datastax/oss/driver/internal/core/metadata/schema/DefaultAggregateMetadata.java

```diff
@@ -114,7 +128,7 @@ public boolean equals(Object other) {
       AggregateMetadata that = (AggregateMetadata) other;
       return Objects.equals(this.keyspace, that.getKeyspace())
           && Objects.equals(this.signature, that.getSignature())
-          && Objects.equals(this.finalFuncSignature, that.getFinalFuncSignature())
+          && Objects.equals(this.finalFuncSignature, that.getFinalFuncSignature().orElse(null))
```

Comment:
Good catch. Should be the same for `initCond` on the next line.
It's a bit unnerving that this is not a compile-time error :(

EDIT -- I checked with a structural search: `Objects.equals($o1$, $o2$)` with a constraint on `o2`'s expression type `java\.util\.Optional`. This is the only occurrence.

Reply:
On my side IntelliJ warns me that an equals comparison is being performed on incompatible types in such situations.